cms/static/js/spec/views/pages/course_outline_spec.js

```diff
@@ -400,6 +400,69 @@ define(['jquery', 'edx-ui-toolkit/js/utils/spec-helpers/ajax-helpers', 'common/j
+                        duplicate_source_locator: xblockLocator,
+                        parent_locator: parentLocator
+                    });
+
+                    // send the response
+                    AjaxHelpers.respondWithJson(requests, {
+                        locator: 'locator-duplicated-xblock'
+                    });
+                };
+
+                it('section can be duplicated', function() {
+                    createCourseOutlinePage(this, mockCourseJSON);
+                    expect(outlinePage.$('.list-sections li.outline-section').length).toEqual(1);
+                    expect(getItemsOfType('section').length, 1);
+                    duplicateXBlockWithSuccess('mock-section', 'mock-course', 'section', 0);
+                    expect(getItemHeaders('section').length, 2);
+                });
+
+                it('subsection can be duplicated', function() {
+                    createCourseOutlinePage(this, mockCourseJSON);
+                    expect(getItemsOfType('subsection').length, 1);
+                    duplicateXBlockWithSuccess('mock-subsection', 'mock-section', 'subsection', 0);
+                    expect(getItemHeaders('subsection').length, 2);
+                });
+
+                it('unit can be duplicated', function() {
+                    createCourseOutlinePage(this, mockCourseJSON);
+                    expandItemsAndVerifyState('subsection');
+                    expect(getItemsOfType('unit').length, 1);
+                    duplicateXBlockWithSuccess('mock-unit', 'mock-subsection', 'unit', 0);
+                    expect(getItemHeaders('unit').length, 2);
+                });
+
+                it('shows a notification when duplicating', function() {
+                    var notificationSpy = EditHelpers.createNotificationSpy();
+                    createCourseOutlinePage(this, mockCourseJSON);
+                    getItemHeaders('section').find('.duplicate-button').first()
+                        .click();
+                    EditHelpers.verifyNotificationShowing(notificationSpy, /Duplicating/);
+                    AjaxHelpers.respondWithJson(requests, {locator: 'locator-duplicated-xblock'});
+                    EditHelpers.verifyNotificationHidden(notificationSpy);
+                });
+
+                it('does not duplicate an xblock upon failure', function() {
+                    var notificationSpy = EditHelpers.createNotificationSpy();
+                    createCourseOutlinePage(this, mockCourseJSON);
+                    getItemHeaders('section').find('.duplicate-button').first()
+                        .click();
+                    EditHelpers.verifyNotificationShowing(notificationSpy, /Duplicating/);
+                    AjaxHelpers.respondWithError(requests);
+                    expect(getItemHeaders('section').length, 2);
+                    EditHelpers.verifyNotificationShowing(notificationSpy, /Duplicating/);
```

Comment:
Don't we show any error message on failure ? I'd like to verify that stat as well.